### ISSUE TITLE
ebiten: a uniform avriable of a vec3 array doesn't work expectedly at least on macOS

### ISSUE THREAD

**Issue by hajimehoshi** [hajimehoshi/ebiten#2463]
### Ebitengine Version

5aecd6762c7f2dda0a9ba82a313d303205ce740e

### Operating System

- [ ] Windows
- [X] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] Xbox
- [ ] Web Browsers

### Go Version (`go version`)

go version go1.19.2 darwin/amd64

### What steps will reproduce the problem?

Run this test

```
func TestShaderVec3Array(t *testing.T) {                                                                                                                                                                                                                                              
        const shader = `package main                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                      
var U [4]vec3                                                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                      
func Fragment(position vec4, texCoord vec2, color vec4) vec4 {                                                                                                                                                                                                                        
        return vec4(U[0].x/255.0, U[1].y/255.0, U[2].z/255.0, U[3].x/255.0)                                                                                                                                                                                                           
}                                                                                                                                                                                                                                                                                     
`                                                                                                                                                                                                                                                                                     
        const w, h = 1, 1                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                      
        dst := ebiten.NewImage(w, h)                                                                                                                                                                                                                                                  
        defer dst.Dispose()                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                      
        s, err := ebiten.NewShader([]byte(shader))                                                                                                                                                                                                                                    
        if err != nil {                                                                                                                                                                                                                                                               
                t.Fatal(err)                                                                                                                                                                                                                                                          
        }                                                                                                                                                                                                                                                                             
        defer s.Dispose()                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                      
        op := &ebiten.DrawRectShaderOptions{}                                                                                                                                                                                                                                         
        op.Uniforms = map[string]any{                                                                                                                                                                                                                                                 
                "U": []float32{                                                                                                                                                                                                                                                       
                        0x24, 0x3f, 0x6a,                                                                                                                                                                                                                                             
                        0x88, 0x85, 0xa3,                                                                                                                                                                                                                                             
                        0x08, 0xd3, 0x13,                                                                                                                                                                                                                                             
                        0x19, 0x8a, 0x2e,                                                                                                                                                                                                                                             
                },                                                                                                                                                                                                                                                                    
        }                                                                                                                                                                                                                                                                             
        dst.DrawRectShader(w, h, s, op)                                                                                                                                                                                                                                               
        if got, want := dst.At(0, 0).(color.RGBA), (color.RGBA{0x24, 0x85, 0x13, 0x19}); !sameColors(got, want, 1) {                                                                                                                                                                  
                t.Errorf("got: %v, want: %v", got, want)                                                                                                                                                                                                                              
        }                                                                                                                                                                                                                                                                             
} 
```

### What is the expected result?

The test passes.

### What happens instead?

```
--- FAIL: TestShaderVec3Array (0.17s)
    shader_test.go:1441: got: {36 163 138 0}, want: {36 133 19 25}
```

### Anything else you feel useful to add?

_No response_

**Comment by TotallyGamerJet** [comment]
+1 since this also fails on arm64:

```
=== RUN   TestShaderVec3Array
    shader_test.go:1133: got: {36 163 138 0}, want: {36 133 19 25}
--- FAIL: TestShaderVec3Array (0.09s)

FAIL
2022-11-20 09:34:11.554 ___TestShaderVec3Array_in_github_com_hajimehoshi_ebiten_v2.test[11844:1142416] [CAMetalLayer nextDrawable] returning nil because allocation failed.
```

**Comment by hajimehoshi** [comment]
Already fixed by 796165492768da892ce897c57b66fbd4c43b640e
